redis.go

```diff
@@ -427,8 +441,12 @@ func (c *baseClient) _process(ctx context.Context, cmd Cmder, attempt int) (bool
 			atomic.StoreUint32(&retryTimeout, 1)
 			return err
 		}
-
-		if err := cn.WithReader(c.context(ctx), c.cmdTimeout(cmd), cmd.readReply); err != nil {
+		readReplyFunc := cmd.readReply
+		// Apply unstable RESP3 search module.
+		if c.opt.UnstableResp3SearchModule && c.isProblematicMethodsOfSearchResp3(ctx, cmd) {
```

Comment:
I would suggest to do `c.opt.UnstableResp3SearchModule` check inside if `c.isProblematicMethodsOfSearchResp3`. And I think it will lead to method rename. Let's keep this unstable functionality as much as possible in the single place to be able to remove this easier in a future.

The method name can be something like `assertStableCommand()`. ANd you may call it only in case if `c.opt.protocol !== 2`